### PR TITLE
wip - add learn/releases page

### PR DESCRIPTION
**Pull request by macqueen**
formatting is pretty messed up still and i need to add some code samples, but wanted to get this up for now. going to fix up a bit more later tonight.

@mitsuhiko are there docs for / did you add deploy creation to sentry-cli?

### CODE REVIEW

File: docs/learn/releases.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+Releases are better with commits
+--------------------------------
+
+If you connect a repository in Sentry, we'll create a webhook to start
+collecting commit data. Once you're ready to create a release, you can
+associate commits with a release either by sending us a list of commit
+ids (shas) with their repos or just by including the current HEAD sha
+and the previous release's HEAD sha. For more information, you can check
+out our `API <https://docs.sentry.io/api/releases/post-organization-releases/>`_ or
```
Comment (mitsuhiko) [inline docs/learn/releases.rst]: We should use internal links here.

**Comment by mitsuhiko** [comment]
@macqueen the deploy integration atm is not finished in sentry-cli due to some flow issues. Going to poke you on slack for this.

File: docs/learn/releases.rst [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+unlocks another feature: Deploy emails. Deploy emails are automatically
+sent to users who have committed to a release that is being deployed.
+
+
+Release Artifacts
+-----------------
+
+Javascript and iOS projects can utilize release artifacts to unminify or
+symbolicate error stack traces. To learn more, please check out our
+:ref:`iOS <cocoa-debug-symbols>` and :ref:`JavaScript <raven-js-sourcemaps>` docs.
```
Comment (macqueen) [inline docs/learn/releases.rst]: note to self, need to add this to sentry-swift

File: docs/learn/releases.rst [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+Releases are used by Sentry to provide you with additional context
+when determining the cause of an issue. You set a release context in
+your SDK, and can either create a release via our API or allow Sentry
+to create one automatically when we receive an event with a new
+release version.
+
+Setting up releases in Sentry is helpful as it will give you access
+to the following features:
+
+#. Marking issues as resolved in the next release
```
Comment (benvinegar) [inline docs/learn/releases.rst]: Should probably add:

```
#. Learn in which release an issue was first introduced / last seen
```

File: docs/learn/releases.rst [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+========
+
+.. note:: Some features of releases are still in beta. If you'd like to check them out, consider joining our early adopter program.
+
+
+What is a Release?
+------------------
+
+Releases are used by Sentry to provide you with additional context
+when determining the cause of an issue. You set a release context in
```
Comment (benvinegar) [inline docs/learn/releases.rst]: > You set a release context in your SDK, and can either create a release via our API or allow Sentry to create one automatically when we receive an event with a new release version.

Ideally we have sections for "set a release context in your SDK" and "create a release via the API" that follow this (before getting into commit data). We could probably clarify this sentence a bit.

Comment (macqueen) [inline docs/learn/releases.rst]: i'll iterate on this later in the day if that's ok with you. i want to get this in so it's not blocking product changes

File: docs/learn/releases.rst (hunk 3 above)
Comment (macqueen) [inline docs/learn/releases.rst]: done

**Comment by jessla** [comment]
@macqueen YASSSSS!!!! thank you for making this page! will definitely be using it to send to people :)

**Pull request merged**